pkg/rpaas/client/client.go

```diff
@@ -0,0 +1,192 @@
+		if err != nil {
+			return err
+		}
+		helperSlice := fp.([]interface{})
+		writeInfo(infoType, helperSlice)
+		return nil
+	}
+
+	bodyString, err := getBodyString(resp)
+	if err != nil {
+		return err
+	}
+	return fmt.Errorf("unexpected status code: body: %v", bodyString)
+}
+
+func (c *RpaasClient) Scale(ctx context.Context, instance string, replicas int32) error {
+	if err := scaleValidate(instance, replicas); err != nil {
+		return err
+	}
+
+	bodyStruct := url.Values{}
+	bodyStruct.Set("quantity", strconv.Itoa(int(replicas)))
+
+	pathName := fmt.Sprintf("/resources/%s/scale", instance)
+	bodyReader := strings.NewReader(bodyStruct.Encode())
+	req, err := c.newRequest("POST", instance, pathName, bodyReader)
+	if err != nil {
+		return err
+	}
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+
+	resp, err := c.do(ctx, req)
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode == http.StatusCreated {
+		return nil
+	}
+
+	bodyString, err := getBodyString(resp)
+	if err != nil {
+		return err
+	}
+	return fmt.Errorf("unexpected status code: body: %v", bodyString)
+}
+
+func (c *RpaasClient) do(ctx context.Context, req *http.Request) (*http.Response, error) {
+	req = req.WithContext(ctx)
+	return c.httpClient.Do(req)
+}
+
+func (c *RpaasClient) newRequest(method, instance, pathName string, body io.Reader) (*http.Request, error) {
+	var url string
+	if c.tsuruAPI != "" {
+		url = fmt.Sprintf("%s/services/%s/proxy/%s?callback=%s",
+			c.tsuruAPI, c.service, instance, pathName)
+
+	} else {
+		url = fmt.Sprintf("%s%s", c.hostAPI, pathName)
+	}
+
+	req, err := http.NewRequest(method, url, body)
+	if err != nil {
+		return nil, err
+	}
+
+	if c.token != "" {
+		req.Header.Add("Authorization", "Bearer "+c.token)
+	}
+
+	return req, nil
+}
+
+func scaleValidate(instance string, replicas int32) error {
+	if instance == "" {
+		return fmt.Errorf("instance can't be nil")
+	}
+
+	if replicas < 0 {
+		return fmt.Errorf("replicas number must be greater or equal to zero")
+	}
+
+	return nil
+}
+
+func getBodyString(resp *http.Response) (string, error) {
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", fmt.Errorf("unable to read body response: %v", err)
+	}
+	defer resp.Body.Close()
+	return string(bodyBytes), nil
+}
+
+// helper table writer functions
+func prepareInfoSlice(data []interface{}) [][]string {
+	dataSlice := [][]string{}
+	for _, mapVal := range data {
+		m := mapVal.(map[string]interface{})
+		target := []string{fmt.Sprintf("%v", m["name"]),
+			fmt.Sprintf("%v", m["description"])}
+		dataSlice = append(dataSlice, target)
+	}
+
+	return dataSlice
+}
+
+func writeInfo(prefix string, data []interface{}) {
+	// flushing stdout
+	fmt.Println()
+
+	dataSlice := prepareInfoSlice(data)
+
+	table := tablewriter.NewWriter(os.Stdout)
+	table.SetRowLine(true)
+	table.SetHeader([]string{prefix, "Description"})
+	for _, v := range dataSlice {
+		table.Append(v)
+	}
+
+	table.Render()
+}
```

Comment:
Drop this copy-and-pasted code...

Reply:
those are the methods that create the ASCII tables from the tsuru target responses...